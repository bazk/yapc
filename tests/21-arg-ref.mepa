     INPP
     AMEM 1
     DSVS R00
R01: ENPR 1 
     AMEM 1
     DSVS R02
R02: NADA 
     CRVL 1, -5
     CRCT 1
     CMME
     DSVF R04
     CRVI 1, -4
     CRVL 1, -5
     SOMA
     ARMI 1, -4
     DSVS R03
R04: NADA 
     CRVI 1, -4
     ARMZ 1, 0
     CRVL 1, -5
     CRCT 1
     SUBT
     CREN 1, 0
     CHPR R01, 1
     CRVL 1, 0
     ARMI 1, -4
     CRVL 1, -5
     CRCT 2
     SUBT
     CRVL 1, -4
     CHPR R01, 1
R03: NADA 
     CRVL 1, -5
     IMPR
     CRVI 1, -4
     IMPR
     DMEM 1
     RTPR 1, 2
R00: NADA 
     CRCT 0
     ARMZ 0, 0
     CRCT 3
     CREN 0, 0
     CHPR R01, 0
     DMEM 1
     PARA
