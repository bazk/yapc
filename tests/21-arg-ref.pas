program exemplo1 (input, output);
var k : integer;
procedure p(    n : integer; var g:integer);
var h : integer;
begin
   if n<1 then
      g:=g+n
   else
   begin
      h:=g;
      p(n-1,h);
      g:=h;
      p(n-2, g)
   end;
   write(n,g)
end;
begin
   k:=0;
   p(3,k)
end.
   
      
