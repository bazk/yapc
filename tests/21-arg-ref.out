00-1-11-10-12-10-1-1-21-23-2